contracts/liquidity_hub/bonding-manager/src/commands.rs

```diff
@@ -0,0 +1,373 @@
+                    Ok(())
+                })
+                .ok_or_else(|| StdError::generic_err("Invalid fee"))?;
+            let denom = &fee.denom;
+            // add the reward to the claimable fees
+            claimable_fees = asset::aggregate_coins(
+                claimable_fees,
+                vec![Coin {
+                    denom: denom.to_string(),
+                    amount: reward,
+                }],
+            )?;
+
+            // modify the epoch to reflect the new available and claimed amount
+            for available_fee in epoch.available.iter_mut() {
+                if available_fee.denom == fee.denom {
+                    available_fee.amount = available_fee.amount.checked_sub(reward)?;
+                }
+            }
+
+            if epoch.claimed.is_empty() {
+                epoch.claimed = vec![Coin {
+                    denom: denom.to_string(),
+                    amount: reward,
+                }];
+            } else {
+                for claimed_fee in epoch.claimed.iter_mut() {
+                    if claimed_fee.denom == fee.denom {
+                        claimed_fee.amount = claimed_fee.amount.checked_add(reward)?;
+                    }
+                }
+            }
+
+            EPOCHS.save(deps.storage, &epoch.id.to_be_bytes(), &epoch)?;
+        }
+    }
+
+    // update the last claimed epoch for the user
+    LAST_CLAIMED_EPOCH.save(deps.storage, &info.sender, &claimable_epochs[0].id)?;
+
+    // Make a message to send the funds to the user
+    let mut messages = vec![];
+    for fee in claimable_fees {
+        messages.push(CosmosMsg::Bank(BankMsg::Send {
+            to_address: info.sender.to_string(),
+            amount: vec![fee.clone()],
+        }));
+    }
+
+    Ok(Response::new()
+        .add_attributes(vec![("action", "claim")])
+        .add_messages(messages))
+}
+
+pub(crate) fn fill_rewards(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+) -> Result<Response, ContractError> {
+    {
+        // Finding the most recent EpochID
+        let most_recent_epoch_id = EPOCHS
+            .keys(deps.storage, None, None, Order::Descending)
+            .next()
+            .unwrap()?;
+
+        let config = CONFIG.load(deps.storage)?;
+        let distribution_denom = config.distribution_denom.clone();
+
+        let mut messages: Vec<CosmosMsg> = vec![];
+        // swap non-whale to whale
+        // Search info funds for LP tokens, LP tokens will contain LP_SYMBOL from lp_common and the string .pair.
+        let mut whale = info
+            .funds
+            .iter()
+            .find(|coin| coin.denom.eq(distribution_denom.as_str()))
+            .unwrap()
+            .to_owned();
+        // Each of these helpers will add messages to the messages vector
+        // and may increment the whale Coin above with the result of the swaps
+        helpers::handle_lp_tokens(&info, &config, &mut messages)?;
+        helpers::swap_coins_to_main_token(
```

Comment:
The funds from the lp withdrawals and the coins swapped are not being added to the epoch. that needs to be handled on a reply like you wrote under